rfc/0075-faster-failover-and-configuration-push.md

```diff
@@ -0,0 +1,315 @@
+like in regular cluster configuration notification, epoch and revision fields are sent as extras.
+
+Note that the magic value for this notification is `ServerRequest` (`0x82`), which is enabled by the `Duplex` (`0x0c`)
+HELLO flag. Additionally, similar to the regular cluster configuration notification, the epoch and revision fields are
+sent as extras.
+
+Once the brief cluster configuration notification is received, it is up to the SDK to decide whether to send a
+`GetClusterConfig` (`0xb5`) request to retrieve the actual configuration body.
+
+In essence, the `ClustermapChangeNotificationBrief` feature only saves network traffic. If
+`DedupeNotMyVbucketClustermap` is not enabled, the number of notifications will be the same as before. However, this
+feature can still be used as a building block to implement a debouncing mechanism. When properly configured, it can help
+reduce the number of requests. Further details on this topic will be covered in the "Library Changes" section.
+
+## Library Changes
+
+### Configuration Push
+
+The previously mentioned `ClustermapChangeNotificationBrief` feature enables the SDK to subscribe all connections for
+configuration updates. These notifications are lightweight and can be deduplicated by the server when the
+`DedupeNotMyVbucketClustermap` option is negotiated.
+
+#### Mixed Clusters
+
+In clusters where there is a mix of nodes with older server versions, meaning that some nodes do not acknowledge
+`ClustermapChangeNotificationBrief`, the respective connection should notify the configuration monitor about its lack of
+support for configuration pushes from the server. As a result, the monitor should utilize the old polling mechanism for
+this particular node instead.
+
+### Enhancements in Handling the `NotMyVbucket` Status
+
+Combination of `DedupeNotMyVbucketClustermap` and `ClustermapChangeNotificationBrief` allows to save traffic by not
+sending configuration, if SDK already seen the same revision, and also sends only pair of `Epoch`/`Revision`. So it is
+up to SDK to initiate configuration update once the non-empty payload returned along with `NotMyVbucket` status code.
+
+Several modifications are required in the SDK:
+1. The retry orchestrator should be able to retry an operation based on configuration updates rather than the timer signal.
+2. The configuration monitor should have the ability to throttle configuration requests due to the following reasons:
+   1. During rebalance, multiple operations may return a `NotMyVbucket` status, triggering a configuration refresh.
+   2. Since `ClustermapChangeNotificationBrief` will cause all connections to subscribe to updates and receive them, it is
+      necessary to account for potential high volumes of updates.
+
+Below is a diagram that illustrates an example of the SDK workflow, where the GET request is waiting for the arrival of
+a new configuration.
+
+```mermaid
```

Comment:
This diagram is a little confusing between step 4 and step 5.  Step 4 doesn't indicate how it is associated with Step 5 (if its associated at all).  IE: Does refreshing the configuration happen in response to the NMV, or is that part of some asynchronous process thats happening.

Reply:
I've updated diagrams